Prototypes/ArmyTaintedGem.cs

```diff
@@ -60,62 +69,39 @@ public void TaintedGem(int quant = 1000)
             mountfrost();
             Bot.Sleep(1500);
             Core.EnsureComplete(7817);
+            q++;
+            Core.Logger($"Quest completed x{q} times");
         }
     }
-    public void Setup()
-    {
-        if (Bot.Map.Name == "boxes")
-        {
-            if (Bot.Player.Username == Bot.Config.Get<string>("player1").ToLower())
-                Core.Jump("Fort2", "Center");
-            else if (Bot.Player.Username == Bot.Config.Get<string>("player2").ToLower())
-                Core.Jump("Closet", "Center");
-            else if (Bot.Player.Username == Bot.Config.Get<string>("player3").ToLower())
-                Core.Jump("Fort1", "Center");
-            else if (Bot.Player.Username == Bot.Config.Get<string>("player4").ToLower())
-                Core.Jump("Boss", "Center");
-            else
-                Core.Jump("Boss", "Center");
-        }
-        else if (Bot.Map.Name == "mountfrost")
-        {
-            if (Bot.Player.Username == Bot.Config.Get<string>("player1").ToLower())
-                Core.Jump("War", "Left");
-            else
-                Core.Jump("War", "Left");
-        }
-        else
-            Core.Logger("Current map not found", messageBox: true, stopBot: true);
-    }
 
     public void boxes()
     {
         Core.Join("boxes");
-        Setup();
-        var task = Bot.Send.PacketSpam("%xt%zm%aggroMon%118891%4%5%8%2%3%1%11%12%13%", "String", Bot.Config.Get<int>("PacketDelay"), cts.Token);
+        if (string.IsNullOrEmpty(Bot.Config.Get<string>("player5").Trim()) && string.IsNullOrEmpty(Bot.Config.Get<string>("player6").Trim()))
+        Army.AggroMonCells("Fort2", "Closet", "Fort1", "Boss");
+        Army.AggroMonStart("boxes");
+        Army.DivideOnCells("Fort2", "Closet", "Fort1", "Boss");
+
         while (!Bot.ShouldExit && (!Core.CheckInventory("Cubes", 500)))
```

Comment:
Not a problem made by you, but there should be a mechanism that causes them to wait when going to the next map, otherwise they might desync.
PR is still valid regardless